### PR TITLE
Build: Migrate @storybook/channel-websocket to strict TS

### PR DESCRIPTION
**Pull request by EDuToit**
Closes #

## What I did
Migrate @storybook/channel-websocket to strict-ts

## Checklist
- [ ] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [ ] Make sure to add/update documentation regarding your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [ ] If this PR should be tested against many or all sandboxes,
      make sure to add the `ci:merged` or `ci:daily` GH label to it.
- [ ] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

File: code/lib/channel-websocket/src/index.ts [hunk 1]
```diff
@@ -20,9 +20,9 @@ interface CreateChannelArgs {
 }
 
 export class WebsocketTransport {
-  private socket: WebSocket;
+  private socket?: WebSocket;
```
Comment (kasperpeulen) [inline code/lib/channel-websocket/src/index.ts:23]: I would prefer inlining the connect function in the constructor, to show TS that it is always defined. As it is private, it is really only used in the constructor anyway.

File: code/lib/channel-websocket/src/index.ts [hunk 2]
```diff
@@ -67,7 +69,9 @@ export class WebsocketTransport {
     };
     this.socket.onmessage = ({ data }) => {
       const event = typeof data === 'string' && isJSON(data) ? parse(data) : data;
-      this.handler(event);
+      if (this.handler) {
+        this.handler(event);
+      }
```
Comment (kasperpeulen) [inline code/lib/channel-websocket/src/index.ts:74]: I think we expect a handler to be set here. Could you do a runtime assertion with `tiny-invariant`?
```ts
invariant(this.handler, "WebsocketTransport handler should be set");
```

**Review by kasperpeulen — CHANGES_REQUESTED** [review]
Thanks! I left some feedback :)

**Review by kasperpeulen — APPROVED** [review]
I rolled bock one socket null check that is not neccesary anymore, LGTM now!

**Comment by EDuToit** [comment]
#22176

**Pull request merged**